Makefile

```diff
@@ -13,29 +13,43 @@ create-env:
 
 .PHONY: install
 install:
-	$(PIP) install --upgrade $(PIP)
+	$(PIP) install --upgrade pip
+	$(PIP) install .
 	$(PIP) install -r $(TEST_DIR)/utils/requirements.txt
+	$(PIP) install pre-commit pytest pytest-cov codecov
 
 .PHONY: test
 test: .cleanCoverage
-	$(PYTEST) $(TEST_DIR)/unit
+	$(PYTEST) $(TEST_DIR)/unit -v
 
 .cleanCoverage:
 	@echo 'cleaning coverage files ...'
-	rm -f .coverage
-	rm -fr htmlcov/
+	rm -rf .coverage htmlcov/
 
 .PHONY: open-report
 open-report: .cleanCoverage
 	$(PYTEST) --cov=$(YARPC) --cov-report=term-missing --cov-report=html
 	open htmlcov/index.html
 
 .PHONY: ci-test
-ci-test:
-	$(PIP) install -r tests/utils/requirements.txt
-	pre-commit run --all-files
-	$(PIP) install .
-	$(PYTEST) --cov --cov-report=xml
+ci-test: lint
+	$(PYTEST) --cov --cov-report=xml -v
+
+.PHONY: flake8
+flake8:
+	@find . -type d -name 'env' -prune -o -name 'docs' -prune -o -name '*.py' -exec flake8 "{}" +
+
+.PHONY: black
+black:
+	@find . -type d -name 'env' -prune -o -name 'docs' -prune -o -name '*.py' -exec black "{}" +
+
+.PHONY: mypy
+mypy:
+	@find . -type d -name 'env' -prune -o -name 'docs' -prune -o -name '*.py' -exec mypy --ignore-missing-imports "{}" +
+
+.PHONY: lint
+lint: flake8 black mypy
+	@echo 'Linting with flake8, black & mypy'
```

Comment:
Forgot to mention, isort should also be included (`isort --check-only`)

Reply:
Another good addition will be `$(PYTHON) setup.py check -rms` instead of `pip install .`.
It will fail though, so comment it out until setup.py is ready

Reply:
Should I leave `pip install .` since `$(PYTHON) setup.py check -rms` will fail?

Reply:
No, I'm going to configure setup.py very soon